src/main/java/seedu/address/logic/commands/ExportCommand.java

```diff
@@ -0,0 +1,223 @@
+import seedu.address.model.person.exceptions.DuplicatePersonException;
+import seedu.address.model.tag.Tag;
+import seedu.address.storage.AddressBookStorage;
+import seedu.address.storage.XmlAddressBookStorage;
+
+/**
+ * @@author shanmu9898
+ * Exports people to an XML file of choice based on tag, index or range
+ */
+public class ExportCommand extends Command {
+
+    public static final String MESSAGE_FAIL = "TeachConnect faced some error while exporting! Please try again!";
+    public static final String MESSAGE_OUT_OF_BOUNDS = "Please check the index bounds!";
+    public static final String MESSAGE_SUCCESS = "Contacts have been successfully exported!";
+    public static final String MESSAGE_RANGE_ERROR = "Please input valid range";
+
+    public static final String COMMAND_WORD = "export";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": exports contacts to the TeachConnect Book based "
+            + "on index, range or tag \n"
+            + "Parameters: "
+            + " [TYPE] "
+            + PREFIX_NAME + " NAME "
+            + PREFIX_RANGE + " RANGE "
+            + PREFIX_TAG_EXPORT + " TAG "
+            + PREFIX_PATH + " PATH \n"
+            + "Example 1: " + COMMAND_WORD + " " + PREFIX_NAME + "Exportfile " + PREFIX_RANGE + "all "
+            + PREFIX_TAG_EXPORT + " friends " + PREFIX_PATH + "/src/main/data \n"
+            + "Example 2: " + COMMAND_WORD + " " + PREFIX_NAME + "Exportfile " + PREFIX_RANGE + "1 "
+            + PREFIX_TAG_EXPORT + " friends " + PREFIX_PATH + "/src/main/data \n"
+            + "Example 3: " + COMMAND_WORD + " " + PREFIX_NAME + "Exportfile " + PREFIX_RANGE + "1,2 "
+            + PREFIX_TAG_EXPORT + " friends " + PREFIX_PATH + "/src/main/data";
+
+
+    private Tag tag;
+    private final String range;
+    private final String path;
+    private AddressBook teachConnectBook;
+    private AddressBookStorage teachConnectStorage;
+    private final String nameOfExportFile;
+
+    /**
+     * Creates an ExportCommand to export the specified {@code Persons}
+     */
+    public ExportCommand(String range, Tag tag, String path, String nameOfExportFile) {
+        requireNonNull(range);
+        requireNonNull(tag);
+        requireNonNull(path);
+        requireNonNull(nameOfExportFile);
+
+        this.range = range;
+        this.path = path;
+        this.tag = tag;
+        this.nameOfExportFile = nameOfExportFile;
+
+        teachConnectBook = new AddressBook();
+    }
+
+    @Override
+    public CommandResult execute() {
+        String[] rangeGiven;
+        try {
+            rangeGiven = handleRange();
+        } catch (IOException e) {
+            return new CommandResult(MESSAGE_RANGE_ERROR);
+        }
+        if (rangeGiven[0].equals("all")) {
+            try {
+                exportAllRange(tag);
+            } catch (DuplicatePersonException e) {
+                throw new AssertionError(MESSAGE_FAIL);
+            }
+        } else {
```

Comment:
You may want to refactor this block of code. Apply SLAP and avoid multiple nested conditional statement.